automation/travisci-test.sh

```diff
@@ -1,41 +1,121 @@
 #!/bin/bash -e
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2020 Red Hat, Inc.
+#
 
+# Do not change this one for debug purposes, it's a guard that functions aren't omitted.
+EXPECTED_FUNCTIONS="test_generate test_binaries test_make test_goveralls_bazel_test test_apidocs test_client_python test_manifests test_olm_verify test_prom_rules_verify"
 
-make generate
-if [[ -n "$(git status --porcelain)" ]] ; then
-    echo "It seems like you need to run 'make generate'. Please run it and commit the changes"
-    git status --porcelain; false
-fi
+last_running_func=""
+trap '{ [[ $last_running_func != "" ]] && echo "Last running function was:" $last_running_func; }' EXIT ERR
 
-if diff <(git grep -c '') <(git grep -cI '') | egrep -v -e 'docs/.*\.png|swagger-ui' -e 'vendor/*' -e 'assets/*' | grep '^<'; then
-    echo "Binary files are present in git repostory."; false
-fi
+# For development you can change this list, but to revert it before merging.
+declare -a functions=( 
+    test_generate 
+    test_binaries
+    test_make 
+    test_goveralls_bazel_test 
+    test_apidocs 
+    test_client_python 
+    test_manifests 
+    test_olm_verify 
+    test_prom_rules_verify    
+)
```

Comment:
I'm sorry, but IMO this is unmaintainable.

In no language, especially not in a scripting one like bash, I expect one to list the functions of executions unless we have in our hand a pluggable program (and that I expect to see in a real programming language and not in a script).

If one wants to add a new functions, there is a need to add it to multiple places to make it kick. That is unreasonable IMO.

It also has little added value, as we expect the relevant action to fail and exit of first error. If you prefer to add `echo` prints at each function, I am fine with that.

Reply:
I will refactor it a bit, have an idea, it was for guarding that a user can run a specific sub group , but not merging the change
(as was discussed with Dan),

there is more value in this PR, for example the latest change that allow to use arg list of the desired functions to run,
allowing parallel run , as discussed with Daniel

Reply:
wanted to remove the guard, so it would be merged, even that this method was discussed with Dan,
and then saw https://github.com/kubevirt/kubevirt/commit/896cdec4f2775bd6e50c9358e5271f4330dcd462
i think this PR supplies a better readable and constructed method, wdyt ?
should we continue ?

Reply:
PTAL